File: 2021/8/part-one.ts
import { readFileSync } from "fs";
import * as utils from "../../utils";

async function main() {
  let result = 0;
  const file = readFileSync(process.argv[2], "utf-8");
  const rows = file
    .trim()
    .split("\n")
    .map((r) => {
      const row = r;
      return row;
    });

  const values = rows.map((r) => {
    const row = r.split(" | ").map((r) => r.split(" "));
    return [row[0], row[1]];
  });

  for (const v of values) {
    for (const o of v[1]) if ([2, 3, 4, 7].includes(o.length)) result++;
  }

  console.log(result);
}

main();

File: 2021/8/part-one.py
import sys

if __name__ == "__main__":
    with open(sys.argv[1], encoding="utf-8") as f:
        rows = [x for x in f]

    values = []
    for r in rows:
        split = r.strip().split(" | ")
        values.append([split[0].strip().split(" "), split[1].strip().split(" ")])

    result = 0
    for v in values:
        result += sum([1 for o in v[1] if len(o) in [2, 3, 4, 7]])

    print(result)
